### PR TITLE
fix for custom events & alerting on failures

### PR DESCRIPTION
**Pull request by nr-kkenney**
# Summary

Turns out the code is structured correctly, but the agent wasn't sending events. The call to `newrelic.recordCustomEvent` is just queueing up the event to be sent, and the agent is asynchronously sending data to new relic. When we hit `process.exit(1)`, it seems to be the case that the event hasn't been sent yet, and the agent wont be able to send it at all.

Solution:
Swapped from using the agent to using the api to record custom events.

Resolves: #930

### CODE REVIEW

File: utils/create-validate-install-plans.js [hunk 1]
```diff
@@ -164,16 +164,16 @@ const createValidateUpdateInstallPlan = async (installPlanFiles) => {
 
 /**
  * @param {boolean} hasFailed if the validation or submission has failed
- * @param {boolean} isDryRun - true for validation, false for submission
+ * @param {Promise<boolean>} isDryRun - true for validation, false for submission
```
Comment (nr-kkenney) [inline utils/create-validate-install-plans.js:167]: that should be return 🙈  whoops

File: utils/newrelic/customEvent.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/**
+ * This module is intended for use only with SSR pages.
+ * For monitoring & creating events for browser pages, use the browser agent instead.
+ */
```
Comment (nr-kkenney) [inline utils/newrelic/customEvent.js:4]: dont need these comments

File: utils/create-validate-install-plans.js [hunk 3]
```diff
@@ -164,16 +164,16 @@ const createValidateUpdateInstallPlan = async (installPlanFiles) => {
...
  */
-const recordCustomNREvent = (hasFailed, isDryRun) => {
+const recordCustomNREvent = async (hasFailed, isDryRun) => {
   const status = hasFailed ? 'failed' : 'success';
   if (isDryRun) {
-    newrelic.recordCustomEvent('ValidateInstallPlans', {
+    await track('ValidateQuickstarts', {
```
Comment (nr-kkenney) [inline utils/create-validate-install-plans.js:172]: these should be install plan events

**Review by josephgregoryii — APPROVED** [review]
looks Grrrreat!

**Review by josephgregoryii — APPROVED** [review]

**Pull request merged**